python/ray/ml/config.py

```diff
@@ -105,6 +115,47 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         bundles = trainer_bundle + worker_bundles
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
+    @classmethod
+    def validate_config(
+        cls,
+        config: Union[ScalingConfig, "ScalingConfigDataClass"],
+        allowed_keys: Iterable[str],
+    ):
+        """
+        Validate config by raising an exception if any of allowed_keys
+        differs from the default value.
+
+        Args:
+            config: Config to check.
+            allowed_keys: ScalingConfigDataClass attribute keys that
+                can have a value different than the default one.
+        """
+        default_config = ScalingConfigDataClass()
+        if not isinstance(config, ScalingConfigDataClass):
+            config = ScalingConfigDataClass(**config)
+
+        allowed_keys = set(allowed_keys)
+
+        bad_allowed_keys = [key for key in allowed_keys if key not in config.__dict__]
+        if bad_allowed_keys:
+            raise KeyError(
+                f"Allowed key(s) {bad_allowed_keys} are not valid. "
+                f"Valid keys: {list(config.__dict__.keys())}"
+            )
+
+        prohibited_keys = set(default_config.__dict__) - allowed_keys
+
+        bad_keys = [
+            key
+            for key in prohibited_keys
+            if config.__dict__[key] != default_config.__dict__[key]
+        ]
+        if bad_keys:
```

Comment:
This currently only checks if the value is set to something other than the default value. But the user can still set a prohibited key to the default value and expect it to be honored. This seems fine for now, but I'm wondering if there's a way to address this case as well.